wsmaster/che-core-api-factory/src/test/java/org/eclipse/che/api/factory/server/urlfactory/DefaultFactoryUrlTest.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2012-2023 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.factory.server.urlfactory;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+import java.util.Optional;
+import org.mockito.testng.MockitoTestNGListener;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+
+/** Testing {@link DefaultFactoryUrl} */
+@Listeners(MockitoTestNGListener.class)
+public class DefaultFactoryUrlTest {
+  @Test(dataProvider = "urlsProvider")
+  public void shouldGetCredentials(String url, String credentials) {
+    // given
+    DefaultFactoryUrl factoryUrl = new DefaultFactoryUrl().withUrl(url);
+    // when
+    Optional<String> credentialsOptional = factoryUrl.getCredentials();
+    // then
+    assertTrue(credentialsOptional.isPresent());
+    assertEquals(credentialsOptional.get(), credentials);
+  }
+
+  @DataProvider(name = "urlsProvider")
+  private Object[][] urlsProvider() {
+    return new Object[][] {
+      {"https://username:password@hostname/path", "username:password"},
+      {"https://token@hostname/path/user/repo/", "token:"},
+      {"http://token@hostname/path/user/repo/", "token:"},
+      {"https://token@dev.azure.com/user/repo/", "token:"},
```

Comment:
```suggestion
      {"https://token@dev.azure.com/user/repo/", "token:"},
      {"https://token@dev.azure.com/user/repo?a=&b=b&c=/.devfile.yaml&api-version=7.0", "token:"},
```
suggestion for also covering URL with fancy parameters